contracts/KetherView.sol

```diff
@@ -1,29 +1,39 @@
 import "./IKetherHomepage.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
 
-contract KetherView {
+library KetherView {
+  struct AdView {
+    address owner;
+    uint x;
+    uint y;
+    uint width;
+    uint height;
+    string link;
+    string image;
+    string title;
+    bool NSFW;
+    bool forceNSFW;
+    bool wrapped;
+  }
   /// allAds is a helper view designed to be called from frontends that want to
   /// display all of the ads with their correct NFT owners.
-  function allAds(IKetherHomepage instance) external view returns (IKetherHomepage.Ad[] memory) {
+  function allAds(address _instanceAddress, address _nftAddress) external view returns (AdView[] memory) {
     // FIXME: Does this actually enumerate all of the ads, or do we need to paginate?
-    uint len = instance.getAdsLength();
-    IKetherHomepage.Ad[] memory ads_ = new IKetherHomepage.Ad[](len);
-
-    for (uint idx=0; idx<len; idx++) {
-      (address owner, uint x, uint y, uint width, uint height, string memory link, string memory image, string memory title, bool NSFW, bool forceNSFW) = instance.ads(idx);
+    AdView[] memory ads_ = new AdView[](IKetherHomepage(_instanceAddress).getAdsLength());
 
-      /* XXX: Fix this
+    for (uint idx=0; idx<ads_.length; idx++) {
+      (address owner, uint x, uint y, uint width, uint height, string memory link, string memory image, string memory title, bool NSFW, bool forceNSFW) = IKetherHomepage(_instanceAddress).ads(idx);
+      bool wrapped = false;
 
       // Is it an NFT already?
-      if (_exists(idx)) {
+      if (owner == _nftAddress) {
```

Comment:
So what fixed that error you were having?

Reply:
This fixed a different error (we don't ave _exists outside of the ERC721 itself).

I just cut variables :)